File: composeApp/src/commonMain/kotlin/felix/livinglink/ui/taskBoard/list/TaskBoardListAddTaskSheet.kt
package felix.livinglink.ui.taskBoard.list

import TaskBoardListScreenLocalizables
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import felix.livinglink.group.Group
import felix.livinglink.ui.common.ModalBottomSheet
import felix.livinglink.ui.common.MultiSelectDropdown

@Composable
fun TaskBoardListAddTaskSheet(
    group: Group?,
    data: TaskBoardListViewModel.Data,
    viewModel: TaskBoardListViewModel
) {
    ModalBottomSheet(
        title = TaskBoardListScreenLocalizables.addTaskSheetTitle(),
        confirmButtonText = TaskBoardListScreenLocalizables.addTaskSheetConfirm(),
        dismissButtonText = TaskBoardListScreenLocalizables.addTaskSheetCancel(),
        onDismiss = viewModel::closeAddTask,
        onConfirm = viewModel::addTask,
        confirmButtonEnabled = viewModel.addTaskConfirmButtonEnabled()
    ) {
        OutlinedTextField(
            value = data.addTaskTitle,
            onValueChange = viewModel::updateAddTaskTitle,
            label = { Text(TaskBoardListScreenLocalizables.addTaskSheetTitleLabel()) },
            singleLine = true,
            modifier = Modifier.fillMaxWidth()
        )

        OutlinedTextField(
            value = data.addTaskDescription,
            onValueChange = viewModel::updateAddTaskDescription,
            label = { Text(TaskBoardListScreenLocalizables.addTaskSheetDescriptionLabel()) },
            minLines = 2,
            modifier = Modifier.fillMaxWidth()
        )

        MultiSelectDropdown(
            title = TaskBoardListScreenLocalizables.addTaskSheetMembersLabel(),
            options = group?.groupMembersSortedByRoleAndName ?: emptyList(),
            selected = data.selectedMemberIds.mapNotNull { id ->
                group?.groupMembersSortedByRoleAndName?.find { it.id == id }
            },
            valueToString = { it.name },
            onSelectionChanged = { user ->
                viewModel.toggleMember(userId = user.id)
            }
        )
    }
}

File: iosApp/iosApp/TaskBoard/List/TaskBoardListAddTaskSheet.swift
//
//  TaskBoardListAddTaskSheet.swift
//  iosApp
//
//  Created by Felix Fischer on 23.07.25.
//  Copyright © 2025 orgName. All rights reserved.
//

import ComposeApp
import SwiftUI

struct TaskBoardListAddTaskSheet: View {
    let group: SharedGroup?
    let data: TaskBoardListViewModel.Data
    let viewModel: TaskBoardListViewModel
    let localizables = TaskBoardListScreenLocalizables()
    @FocusState private var focusedField: Field?

    var body: some View {
        ModalBottomSheet(
            title: localizables.addTaskSheetTitle.localized,
            onDismiss: viewModel.closeAddTask,
            onConfirm: viewModel.addTask,
            confirmButtonEnabled: viewModel.addTaskConfirmButtonEnabled()
        ) {
            VStack(spacing: DesignSystem.Spacing.betweenElements) {
                Section {
                    CustomTextField(
                        label: localizables.addTaskSheetTitleLabel.localized,
                        text: data.addTaskTitle,
                        onChange: viewModel.updateAddTaskTitle(title:)
                    )
                    .textFieldStyle(DesignSystem.CustomTextFieldStyle())
                    .focused($focusedField, equals: .title)
                    .submitLabel(.next)
                    .onSubmit {
                        focusedField = .description
                    }

                    CustomTextEditor(
                        label: localizables.addTaskSheetDescriptionLabel.localized,
                        text: data.addTaskDescription,
                        onChange: viewModel.updateAddTaskDescription(description:)
                    )
                    .focused($focusedField, equals: .description)
                }

                CustomMultiPicker(
                    title: localizables.addTaskSheetMembersLabel.localized,
                    options: group?.groupMembersSortedByRoleAndName.map { $0.id } ?? [],
                    selected: data.selectedMemberIds,
                    valueToString: { id in
                        group?.groupMemberIdsToName[id] ?? ""
                    },
                    label: {
                        Text(localizables.addTaskSheetMembersLabel.localized)
                    },
                    onSelectionChanged: { selected in
                        viewModel.toggleMember(userId: selected)
                    }
                )
                .padding(.vertical, DesignSystem.Padding.small)
            }
            .onAppear { focusedField = .title }
        }
    }

    private enum Field {
        case title
        case description
    }
}
